frontend/src/store/connection/reducers.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  ConnectionAction,
+  ConnectionState,
+  BACKEND_CONNECTION_UP,
+  BACKEND_CONNECTION_DOWN,
+  BACKEND_HEARTBEAT,
+} from './types';
+
+const initialState: ConnectionState = {
+  backendHeartbeat: new Date(),
+  backendConnection: false,
+  lastBackendConnectionTime: null,
+};
+
+export function connectionReducer(state = initialState, action: ConnectionAction): ConnectionState {
+  switch (action.type) {
+    case BACKEND_CONNECTION_DOWN:
+      return { ...state, backendConnection: false };
+    case BACKEND_HEARTBEAT:
+      // TODO: Redux recommends **against** using new date inside reducers; instead
```

Comment:
Let's move this TODO comment to line 29 in the `BACKEND_CONNECTION_UP` case. Even better is if we could create the `new Date()` in `action.ts`'s `establishedBackendConnection()` action creator like what `receivedBackendHeartbeat` does right now.